deepchecks/tabular/metric_utils/scorers.py

```diff
@@ -291,26 +290,44 @@ def _run_score(self, model, dataset: 'tabular.Dataset'):
                 f'not provided. Please use a model with predict_proba method or '
                 f'manually provide predicted probabilities to the check. '
                 f'{doclink_str}')
-        if self.possible_classes is not None:
-            updated_model, transformed_label_col = self._wrap_classification_model(model, dataset.label_col)
-            return self.scorer(updated_model, dataset.features_columns, transformed_label_col)
+        if self.model_classes is not None:
+            updated_model = self._wrap_classification_model(model)
+            if len(self.model_classes) == 2:
+                updated_label_col = label_col.map({self.model_classes[0]: 0, self.model_classes[1]: 1})
+                scores = self.scorer(updated_model, data, updated_label_col)
+            else:
+                label = _transform_to_multi_label_format(np.array(label_col), self.model_classes)
+                scores = self.scorer(updated_model, data, label)
+
+            # The scores returned are for the observed classes but we want scores of the observed classes
+            if isinstance(scores, t.Sized):
+                if len(scores) != len(self.model_classes):
+                    raise errors.DeepchecksValueError(
+                        f'Scorer returned {len(scores)} scores, but model contains '
+                        f'{len(self.model_classes)} classes. Can\'t proceed')
+                scores = dict(zip(self.model_classes, scores))
+                # Add classes which been seen in the data but are not known to the model
+                scores.update({cls: np.nan for cls in set(self.observed_classes) - set(self.model_classes)})
+            return scores
         else:
-            return self.scorer(model, dataset.features_columns, dataset.label_col)
+            return self.scorer(model, data, label_col)
 
     def __call__(self, model, dataset: 'tabular.Dataset'):
         """Run score with labels null filtering."""
-        return self._run_score(model, self.filter_nulls(dataset))
+        dataset_without_nulls = self.filter_nulls(dataset)
```

Comment:
Optional: see if you can easily remove this (time consuming func)

Reply:
no